### PR TITLE
Add implementations for random_hue

### PR DESCRIPTION
**Pull request by shashaka**
I have implemented the RandomHue image processing feature. Please let me know if any further corrections or improvements are required.

here is my [gist](https://colab.research.google.com/drive/1WrdTiyBvOCWFERooeKE5ddT-Y_HctKAA?usp=sharing)

### CODE REVIEW

**Comment by fchollet** [comment]
Thanks for the PR!

It looks like the numerical sensitivity of the tests should be reduced slightly:

```
FAILED keras/src/layers/preprocessing/image_preprocessing/random_hue_test.py::RandomHueTest::test_random_hue_no_change_with_zero_factor - AssertionError: 
Not equal to tolerance rtol=1e-06, atol=0.0001
None
Mismatched elements: 1 / 150528 (0.000664%)
Max absolute difference among violations: 0.00012398
Max relative difference among violations: 5.39030404e-06
```

Also, it looks like running the layer in a tf.data pipeline still leads to torch tensors getting created, which should be the case -- everything should TF graph mode in this case:

```
___________________ RandomHueTest.test_tf_data_compatibility ___________________

self = 

    def test_tf_data_compatibility(self):
        data_format = backend.config.image_data_format()
        if data_format == "channels_last":
            input_data = np.random.random((2, 8, 8, 3))
        else:
            input_data = np.random.random((2, 3, 8, 8))
        layer = layers.RandomHue(
            factor=0.5, value_range=[0, 1], data_format=data_format, seed=1337
        )
    
>       ds = tf_data.Dataset.from_tensor_slices(input_data).batch(2).map(layer)

keras/src/layers/preprocessing/image_preprocessing/random_hue_test.py:64: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/tmpfs/venv/lib/python3.9/site-packages/tensorflow/python/data/ops/dataset_ops.py:2341: in map
    return map_op._map_v2(
/tmpfs/venv/lib/python3.9/site-packages/tensorflow/python/data/ops/map_op.py:43: in _map_v2
    return _MapDataset(
/tmpfs/venv/lib/python3.9/site-packages/tensorflow/python/data/ops/map_op.py:157: in __init__
    self._map_func = structured_function.StructuredFunctionWrapper(
/tmpfs/venv/lib/python3.9/site-packages/tensorflow/python/data/ops/structured_function.py:265: in __init__
    self._function = fn_factory()
/tmpfs/venv/lib/python3.9/site-packages/tensorflow/python/eager/polymorphic_function/polymorphic_function.py:1251: in get_concrete_function
    concrete = self._get_concrete_function_garbage_collected(*args, **kwargs)
/tmpfs/venv/lib/python3.9/site-packages/tensorflow/python/eager/polymorphic_function/polymorphic_function.py:1221: in _get_concrete_function_garbage_collected
    self._initialize(args, kwargs, add_initializers_to=initializers)
/tmpfs/venv/lib/python3.9/site-packages/tensorflow/python/eager/polymorphic_function/polymorphic_function.py:696: in _initialize
    self._concrete_variable_creation_fn = tracing_compilation.trace_function(
/tmpfs/venv/lib/python3.9/site-packages/tensorflow/python/eager/polymorphic_function/tracing_compilation.py:178: in trace_function
    concrete_function = _maybe_define_function(
/tmpfs/venv/lib/python3.9/site-packages/tensorflow/python/eager/polymorphic_function/tracing_compilation.py:283: in _maybe_define_function
    concrete_function = _create_concrete_function(
/tmpfs/venv/lib/python3.9/site-packages/tensorflow/python/eager/polymorphic_function/tracing_compilation.py:310: in _create_concrete_function
    traced_func_graph = func_graph_module.func_graph_from_py_func(
/tmpfs/venv/lib/python3.9/site-packages/tensorflow/python/framework/func_graph.py:1059: in func_graph_from_py_func
    func_outputs = python_func(*func_args, **func_kwargs)
/tmpfs/venv/lib/python3.9/site-packages/tensorflow/python/eager/polymorphic_function/polymorphic_function.py:599: in wrapped_fn
    out = weak_wrapped_fn().__wrapped__(*args, **kwds)
/tmpfs/venv/lib/python3.9/site-packages/tensorflow/python/data/ops/structured_function.py:231: in wrapped_fn
    ret = wrapper_helper(*args)
/tmpfs/venv/lib/python3.9/site-packages/tensorflow/python/data/ops/structured_function.py:161: in wrapper_helper
    ret = autograph.tf_convert(self._func, ag_ctx)(*nested_args)
/tmpfs/venv/lib/python3.9/site-packages/tensorflow/python/autograph/impl/api.py:690: in wrapper
    return converted_call(f, args, kwargs, options=options)
/tmpfs/venv/lib/python3.9/site-packages/tensorflow/python/autograph/impl/api.py:377: in converted_call
    return _call_unconverted(f, args, kwargs, options)
/tmpfs/venv/lib/python3.9/site-packages/tensorflow/python/autograph/impl/api.py:459: in _call_unconverted
    return f(*args, **kwargs)
keras/src/layers/preprocessing/tf_data_layer.py:43: in __call__
    outputs = super().__call__(inputs, **kwargs)
keras/src/utils/traceback_utils.py:113: in error_handler
    return fn(*args, **kwargs)
keras/src/layers/layer.py:905: in __call__
    outputs = super().__call__(*args, **kwargs)
/tmpfs/venv/lib/python3.9/site-packages/torch/nn/modules/module.py:1736: in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
/tmpfs/venv/lib/python3.9/site-packages/torch/nn/modules/module.py:1747: in _call_impl
    return forward_call(*args, **kwargs)
keras/src/backend/torch/layer.py:44: in forward
    return Operation.__call__(self, *args, **kwargs)
keras/src/utils/traceback_utils.py:113: in error_handler
    return fn(*args, **kwargs)
keras/src/ops/operation.py:54: in __call__
    return self.call(*args, **kwargs)
keras/src/layers/preprocessing/image_preprocessing/base_image_preprocessing_layer.py:130: in call
    transformation = self.get_random_transformation(data, training=training)
keras/src/layers/preprocessing/image_preprocessing/random_hue.py:123: in get_random_transformation
    -self.backend.numpy.ones_like(invert),
keras/src/backend/tensorflow/numpy.py:1691: in ones_like
    return tf.ones_like(x, dtype=dtype)
/tmpfs/venv/lib/python3.9/site-packages/tensorflow/python/ops/weak_tensor_ops.py:88: in wrapper
    return op(*args, **kwargs)
/tmpfs/venv/lib/python3.9/site-packages/tensorflow/python/util/traceback_utils.py:153: in error_handler
    raise e.with_traceback(filtered_tb) from None
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = tensor([0.8543], device='cuda:0'), dtype = None

    def __array__(self, dtype=None):
        if has_torch_function_unary(self):
            return handle_torch_function(Tensor.__array__, (self,), self, dtype=dtype)
        if dtype is None:
>           return self.numpy()
E           TypeError: can't convert cuda:0 device type tensor to numpy. Use Tensor.cpu() to copy the tensor to host memory first.

/tmpfs/venv/lib/python3.9/site-packages/torch/_tensor.py:1149: TypeError
```

This is likely due to the fact that you're using `keras.random` directly (which will use torch), e.g.

```
invert = keras.random.uniform((1,), seed=self.seed_generator)
```

In reality you should do:

```
if seed is None:
    seed = self._get_seed_generator(self.backend._backend)
invert = self.backend.random.uniform((1,), seed=seed)
```

**Review by fchollet — COMMENTED** [review]
The code is looking good! 👍

File: keras/src/layers/preprocessing/image_preprocessing/random_hue.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+             `factor` controls the extent to which the
+            image hue is impacted. `factor=0.0` makes this layer perform a
+            no-op operation, while a value of 1.0 performs the most aggressive
+            contrast adjustment available. If a tuple is used, a `factor` is
+            sampled between the two values for every image augmented. If a
+            single float is used, a value between `0.0` and the passed float is
+            sampled. In order to ensure the value is always the same, please
+            pass a tuple with two identical floats: `(0.5, 0.5)`.
+        value_range: the range of values the incoming images will have.
+            Represented as a two number tuple written [low, high]. This is
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/layers/preprocessing/image_preprocessing/random_hue.py]: I added backticks around the code. Thanks.

**Comment by shashaka** [comment]
I've corrected the failed test cases and updated the description. Please let me know if any further corrections are needed.

File: keras/src/layers/preprocessing/image_preprocessing/random_hue.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+            sampled between the two values for every image augmented. If a
+            single float is used, a value between `0.0` and the passed float is
+            sampled. In order to ensure the value is always the same, please
+            pass a tuple with two identical floats: `(0.5, 0.5)`.
+        value_range: the range of values the incoming images will have.
+            Represented as a two-number tuple written `[low, high]`. This is
+            typically either `[0, 1]` or `[0, 255]` depending on how your
+            preprocessing pipeline is set up.
+        seed: Integer. Used to create a random seed.
+
```
Comment (fchollet) [inline keras/src/layers/preprocessing/image_preprocessing/random_hue.py:52]: Add: "Example:"

**Review by fchollet — APPROVED** [review]
Looking good!

File: keras/src/layers/preprocessing/image_preprocessing/random_hue.py (hunk 2 above)
Comment (shashaka) [reply keras/src/layers/preprocessing/image_preprocessing/random_hue.py:52]: I will update description by another PR. Thanks.

**Pull request merged**